### PR TITLE
fix: Fix AccountFoundries and add test

### PR DESCRIPTION
**Pull request by howjmay**

### CODE REVIEW

File: packages/vm/core/accounts/impl_views.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func viewAccountFoundries(ctx isc.SandboxView) dict.Dict {
 	foundries := accountFoundriesMapR(ctx.StateR(), account)
 	ret := dict.New()
 	foundries.MustIterate(func(k []byte, v []byte) bool {
-		ret.Set(kv.Key(k), v)
+		ret.Set(kv.Key(k), []byte{0xff})
```
Comment (dessaya) [inline packages/vm/core/accounts/impl_views.go]: Just curious, why this change?

File: contracts/wasm/corecontracts/test/core_accounts_test.go [hunk 2]
```diff
@@ -257,6 +257,52 @@ func TestFoundryModifySupply(t *testing.T) {
...
+		require.Equal(t, []byte{0xff}, f.Results.Foundries().GetBytes(foundry.SN()).Value())
+	}
+}
+
+func TestAccountNFTAmount(t *testing.T) {}
+
+func TestAccountNFTAmountInCollection(t *testing.T) {}
+
+func TestAccountNFTsInCollection(t *testing.T) {}
+
```
Comment (dessaya) [inline contracts/wasm/corecontracts/test/core_accounts_test.go]: Are these left empty intentionally?

File: packages/vm/core/accounts/impl_views.go (hunk 1 above)
Comment (howjmay) [reply packages/vm/core/accounts/impl_views.go]: I am referring the wiki
https://wiki.iota.org/shimmer/smart-contracts/guide/core_concepts/core_contracts/accounts#accountfoundriesa-agentid

File: contracts/wasm/corecontracts/test/core_accounts_test.go (hunk 2 above)
Comment (howjmay) [reply contracts/wasm/corecontracts/test/core_accounts_test.go]: Yes. I will finish them in another PR

File: packages/vm/core/accounts/impl_views.go (hunk 1 above)
Comment (dessaya) [reply packages/vm/core/accounts/impl_views.go]: But the `v` value already should contain `0xff`. If it does not, then we have a bug somewhere else.

File: contracts/wasm/corecontracts/test/core_accounts_test.go (hunk 2 above)
Comment (dessaya) [reply contracts/wasm/corecontracts/test/core_accounts_test.go]: Can you remove them in this PR? Or add a comment explaining why there are empty? Otherwise it may be confusing.

File: packages/vm/core/accounts/impl_views.go (hunk 1 above)
Comment (dessaya) [reply packages/vm/core/accounts/impl_views.go]: Just to clarify, I think it is better to leave `v` here because in the future we might decide how a `true` value is encoded (it happened recently).

Comment (howjmay) [reply packages/vm/core/accounts/impl_views.go]: It was `0x1` there. Let me check what cause it

File: contracts/wasm/corecontracts/test/core_accounts_test.go (hunk 2 above)
Comment (howjmay) [reply contracts/wasm/corecontracts/test/core_accounts_test.go]: ok

File: packages/vm/core/accounts/impl_views.go (hunk 1 above)
Comment (howjmay) [reply packages/vm/core/accounts/impl_views.go]: done
